### PR TITLE
Spritelab set background as block backwards compatibility

### PR DESCRIPTION
**Pull request by JillianK**
Adds any sprites from the default sprite list that are missing from a project's initial sprites (generally because the defaultSprites have changed since its creation)
Also adds extra logic to check for missing sprites after filtering by category to prevent the new sprite block and the background block from breaking.

This is necessary before we add the block to the toolbox.

## Links

- [Slack conversation about the issue](https://codedotorg.slack.com/archives/C1B3PNDL7/p1604690860297000)
- [PR that introduced this potential problem](https://github.com/code-dot-org/code-dot-org/pull/36579)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/p5lab/spritelab/blocks.js [hunk 1]
```diff
@@ -224,6 +229,7 @@ const customInputTypes = {
           }
         ];
       }
+      console.log('about to do things');
```
Comment (breville) [inline apps/src/p5lab/spritelab/blocks.js]: Stray logging?

Comment (JillianK) [reply apps/src/p5lab/spritelab/blocks.js]: oops yes

File: apps/src/p5lab/spritelab/blocks.js [hunk 2]
```diff
@@ -40,6 +40,11 @@ function animations(areBackgrounds) {
         return [url, `"${animation.name}"`];
       }
     });
+  // In case either all backgrounds or all costumes are missing and we request them
+  if (results.length === 0) {
+    return [['sprites missing', 'null']];
+  }
```
Comment (breville) [inline apps/src/p5lab/spritelab/blocks.js]: What's the user experience here?  is it worth noting in the comment?

Comment (JillianK) [reply apps/src/p5lab/spritelab/blocks.js]: the block will show up but the dropdown of backgrounds or sprites will be empty and they'll just have to add sprites themselves using the "more" button. I can add it to the comment!

**Review by breville — APPROVED** [review]

**Pull request merged**